stdlib/public/core/OutputStream.swift

```diff
@@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
+    if string.isEmpty { return }
+
+    if string._core.isASCII {
+      defer { _fixLifetime(string) }
+
+      let result = _swift_stdlib_fwrite_stdout(
+        UnsafePointer(string._core.startASCII), string._core.count, 1)
+      if result != 1 {
+        fatalError("fwrite() returned \(result), expected 1")
```

Comment:
I think current behavior is a bug, since it results in data loss.

If you were to print to a file, and the operation couldn't complete because of I/O error, sure you wouldn't want the issue to silently go unnoticed?  So why is stdout different?  It can also be redirected to a file.